core/drivers/pl022_spi.c

```diff
@@ -184,6 +194,14 @@ static void pl022_txrx8(struct spi_chip *chip, uint8_t *wdat, uint8_t *rdat,
 			if (read8(pd->base + SSPSR) & SSPSR_RNE)
 				/* rx 1 packet */
 				rdat[j++] = read8(pd->base + SSPDR);
+			else
+				cnt++;
+
+			if (cnt > SSP_LOOP_TIMEOUT) {
```

Comment:
@jforissier You're right! I tested on HiKey with what only seemed like a sufficiently big enough loop count. Time functions are not available and even if they are, it might not easy to find a time value that works for all platforms.

I've pushed a change that checks for register busy status instead, which is what I should have done in the first place, so thanks for catching this!